src/handlers/CrossMarginHandler.sol

```diff
@@ -283,6 +290,30 @@ contract CrossMarginHandler is OwnableUpgradeable, ReentrancyGuardUpgradeable, I
     return _orderId;
   }
 
+  /// @notice Calculate new trader balance after transfer collateral token.
+  /// @dev This uses to calculate new trader balance when they tranferring token as collateral.
+  /// @param _primaryAccount Trader's primary address from trader's wallet.
+  /// @param _subAccountIdFrom Trader's Sub-Account Id to withdraw from.
+  /// @param _subAccountIdTo Trader's Sub-Account Id to deposit to.
+  /// @param _token Token that's withdrawn as collateral.
+  /// @param _amount Token withdrawing amount.
+  function transferCollateralSubAccount(
+    address _primaryAccount,
+    uint8 _subAccountIdFrom,
+    uint8 _subAccountIdTo,
+    address _token,
+    uint256 _amount
+  ) external nonReentrant onlyAcceptedToken(_token) {
+    if (_amount == 0) revert ICrossMarginHandler_BadAmount();
+    if (_subAccountIdFrom == _subAccountIdTo) revert ICrossMarginHandler_SelfTransfer();
+    // SLOAD
+    CrossMarginService _crossMarginService = CrossMarginService(crossMarginService);
+
+    _crossMarginService.transferCollateral(_primaryAccount, _primaryAccount, _subAccountIdFrom, _subAccountIdTo, _token, _amount);
+
+    emit LogTransferCollateralSubAccount(_primaryAccount, _subAccountIdFrom, _subAccountIdTo, _token, _amount);
+  }
```

Comment:
Was looking through this new PR out of interest, and was wondering if this should be added:

```solidity
require(msg.sender == _primaryAccount);
```

or an equivalent suggestion is to remove `_primaryAccount` and only use `msg.sender`

Reply:
Thanks for the catch!

@chenciao8 Please remove primaryAccount and use msg.sender

Reply:
Thanks. Fixed.